lib/Sema/TypeCheckAvailability.cpp

```diff
@@ -2509,12 +2509,19 @@ bool AvailabilityWalker::diagAvailability(const ValueDecl *D, SourceRange R,
       return true;
   }
 
-  if (FragileKind)
-    if (R.isValid())
+  if (FragileKind) {
+    if (R.isValid()) {
       if (TC.diagnoseInlinableDeclRef(R.Start, D, DC, *FragileKind,
                                       TreatUsableFromInlineAsPublic))
         return true;
 
+      auto *SF = cast<SourceFile>(DC->getModuleScopeContext());
+      auto *M = D->getDeclContext()->getParentModule();
+      if (SF->getParentModule() != M)
+        SF->markUsableFromInlineImport(M);
```

Comment:
The owning module might not be imported directly (for example, AppKit re-exports Foundation, and `_exported` imports elsewhere in the module are also visible). You'll need something a little more powerful here—it might just come down to adding additional imports.

Reply:
Oh, adding an additional import should be OK as long as we only do it once.

Reply:
I'm not sure what you mean by "only do it once". It'll happen once per indirect import.